### ISSUE TITLE
ensure content lengths where possible

### ISSUE THREAD

**Issue by cainus** [cainus/percolator#69]

**Comment by lboening** [comment]
Yes, there seem to be some clients like Internet Explorer 10 and 11 that would prefer a content-length value.

**Comment by cainus** [comment]
What sort of problems are you seeing with those browsers?  Thanks for the
extra info!

**Comment by lboening** [comment]
We are user Dancer (perl language) to deliver our API. The Dancer program can act as a client to talk to an internal API which we have written in Microsofts Web API 2.0. We are able stream a file through it and we set the Transfer-Method to Chunked. Well, it seems to cause more problems (transfer-method: chunked) than it solves. The entity body really should have a "Content-Length" header and a value computed from the entity body. All is well when we do it that way. All clients (chrome, IE 8-11, Safari, Opera) love to receive a content-length header and a value!  It ALWAYS works (well, maybe it doesn't).  Internet Explorer 10-11, however seems to get "balky" on just receiving a transfer-method: chunked. We will probably have to abandon transfer-method: chunked on our API because of this glitch. 

In my searching, I found this project and I am going to try it. Just to prevent errors, this feature seems to be a great idea!

So,if there was an upvote button, I would upvote "Content-Lengths"!
